### PR TITLE
Clean up LBFGS

### PR DESCRIPTION
**Pull request by vladimir-ch**
@btracey PTAL

### CODE REVIEW

File: lbfgs.go [hunk 1]
```diff
@@ -1,52 +1,61 @@
...
 
-// LBFGS implements the limited-memory BFGS algorithm. While the normal BFGS algorithm
-// makes a full approximation to the inverse hessian, LBFGS instead approximates the
-// hessian from the last Store optimization steps. The Store parameter is a tradeoff
-// between cost of the method and accuracy of the hessian approximation.
-// LBFGS has a cost (both in memory and time) of O(Store * inputDimension).
-// Since BFGS has a cost of O(inputDimension^2), LBFGS is more appropriate
-// for very large problems. This "forgetful" nature of LBFGS may also make it perform
-// better than BFGS for functions with Hessians that vary rapidly spatially.
+// LBFGS implements the limited-memory BFGS optimization method.
```
Comment (btracey) [inline lbfgs.go]: why method instead of algorithm?

File: lbfgs.go [hunk 2]
```diff
@@ -1,52 +1,61 @@
...
+//
+// While the normal BFGS algorithm makes a full approximation to the inverse
+// Hessian, LBFGS instead approximates the Hessian from the last Store
+// iterations. The Store parameter is a tradeoff between cost of the method and
```
Comment (btracey) [inline lbfgs.go]: between storage cost. Strictly it's both, but storage is the important dimension. We can leave the last line to highlight it's both.

Comment (btracey) [inline lbfgs.go]: I guess that's not true. I would delete this last sentence here and move the whole discussion into the next paragraph. This section says what it does and the next one says why

File: lbfgs.go [hunk 3]
```diff
@@ -1,52 +1,61 @@
...
+// accuracy of the Hessian approximation. LBFGS has a cost (both in memory and
+// time) of O(Store * dimension).
 //
-// If Store is 0, Store is defaulted to 15.
-// A Linesearcher for LBFGS must satisfy the strong Wolfe conditions at every
-// iteration. If Linesearcher == nil, an appropriate default is chosen.
+// Since BFGS has a cost of O(dimension^2), LBFGS is more appropriate for very
```
Comment (btracey) [inline lbfgs.go]: I would flip the sentence

LBFGS is more appropriate than BFGS for large dimensional problems as the cost of LBFGS scales as O(Store * dim) while BFGS scales with O(dim^2).

File: lbfgs.go [hunk 4]
```diff
@@ -1,52 +1,61 @@
...
+// large problems. The "forgetful" nature of LBFGS may also make it perform
+// better than BFGS for functions with Hessians that vary rapidly spatially.
 type LBFGS struct {
+	// Linesearcher selects suitable steps along the descent direction.
+	// Accepted steps should satisfy the strong Wolfe conditions.
+	// If Linesearcher == nil, a reasonable default will be chosen.
 	Linesearcher Linesearcher
-	Store        int // how many past iterations to store
+	// Store is the size of the limited-memory storage.
+	// If Store is 0, it will be defaulted to 15.
```
Comment (btracey) [inline lbfgs.go]: We should choose one of is and ==. @kortschak  the preferred is english in comments, right?

**Comment by btracey** [comment]
LGTM

File: lbfgs.go (hunk 4 above)
Comment (kortschak) [inline lbfgs.go]: English over symbols if possible.

File: lbfgs.go (hunk 1 above)
Comment (vladimir-ch) [inline lbfgs.go]: Just for consistency with the other optimizers which all use the word "method".

**Comment by vladimir-ch** [comment]
@btracey Could you please take one last look? I changed the LBFGS comment and I deleted the tradeoff comment. I considered deleting the word "spatially" because from the point of view of LBFGS there is no other than spatial variation but in the end I left it there.

**Comment by btracey** [comment]
LGTM

**Pull request merged**